apps/src/gamelab/GameLab.js

```diff
@@ -826,38 +826,132 @@ GameLab.prototype.onP5ExecutionStarting = function () {
 };
 
 /**
- * This is called while this.gameLabP5 is in the preload phase. We initialize
- * the interpreter, start its execution, and call the user's preload function.
+ * This is called while this.gameLabP5 is in the preload phase. Do the following:
  *
- * @return {Boolean} whether or not the preload has completed
+ * - load animations into the P5 engine
+ * - initialize the interpreter
+ * - start its execution
+ * - (optional) execute global code
+ * - call the user's preload function
+ *
+ * @return {Boolean} whether or not the preload has completed.  If we return
+ *         FALSE here, P5 will internally increment a preload count; calling
+ *         notifyPreloadPhaseComplete is then necessary to continue starting
+ *         the game.
  */
 GameLab.prototype.onP5Preload = function () {
-  this.gameLabP5.preloadAnimations(this.studioApp_.reduxStore.getState().animationList);
-
-  this.initInterpreter();
-  // And execute the interpreter for the first time:
-  if (this.JSInterpreter && this.JSInterpreter.initialized()) {
-    // Start executing the interpreter's global code as long as a setup() method
-    // was provided. If not, we will skip running any interpreted code in the
-    // preload phase and wait until the setup phase.
-    this.preloadInProgress = true;
-    if (this.globalCodeRunsDuringPreload) {
-      this.JSInterpreter.executeInterpreter(true);
-      this.interpreterStarted = true;
+  let isReadyNow = false;
+  Promise.all([
+      this.preloadAnimations_(),
+      this.runPreloadEventHandler_()
+  ]).then(() => {
+    isReadyNow = true;
+    this.gameLabP5.notifyPreloadPhaseComplete();
+  });
+  return isReadyNow;
```

Comment:
Hmm... For some reason, I was under the impression that when both `preloadAnimations_` and `runPreloadEventHandler_` resolve immediately (and I've checked, they do both call `resolve()` before hitting the return statement), that the `then` clause here would also run before the return statement.  It seems that's not the case though; so perhaps I should just always return false?  That would somewhat simplify the logic, on the whole.